### PR TITLE
Enable more warnings in the build system

### PR DESCRIPTION
**Pull request by axic**
Part of #6905.

### CODE REVIEW

**Comment by axic** [comment]
Weird, my macos clang wasn't as pedantic as CI's :wink:

File: libsolidity/codegen/LValue.cpp [hunk 1]
```diff
@@ -339,7 +339,6 @@ void StorageItem::storeValue(Type const& _sourceType, SourceLocation const& _loc
 				{
 					solAssert(sourceType.location() == DataLocation::Memory, "");
 					// stack layout: source_ref target_ref
-					TypePointer sourceMemberType = sourceType.memberType(member.name);
```
Comment (axic) [inline libsolidity/codegen/LValue.cpp:343]: This is already done on line 327 and this was a dupliacte.

**Comment by axic** [comment]
I'm not proud of the renaming for avoiding shadowing, but it did spot one mistake.

**Comment by axic** [comment]
Shadowing is super annoying as it depends on compiler version, none of these failures are shown on macos.

**Comment by axic** [comment]
@leonardoalt you can also check this monster

**Review by Marenz — CHANGES_REQUESTED** [review]
Apart from the failing tests this LGTM

**Review by erak — CHANGES_REQUESTED** [review]
Builds are failing. Did not take a deeper look at the specific issues, though.

**Comment by ekpyron** [comment]
Do we *really* want to disallow shadowing :-)? If we really want to, I'd manage to cope, but to me personally that feels much more annoying than helpful... what mistake did that reveal?

File: libsolidity/interface/CompilerStack.cpp [hunk 2]
```diff
@@ -1501,8 +1501,8 @@ Json::Value CompilerStack::gasEstimates(string const& _contractName) const
 			FunctionType type(*it);
 			string sig = it->name() + "(";
 			auto paramTypes = type.parameterTypes();
-			for (auto it = paramTypes.begin(); it != paramTypes.end(); ++it)
-				sig += (*it)->toString() + (it + 1 == paramTypes.end() ? "" : ",");
+			for (auto params_it = paramTypes.begin(); params_it != paramTypes.end(); ++params_it)
```
Comment (leonardoalt) [inline libsolidity/interface/CompilerStack.cpp]: Style should be camel

File: libyul/optimiser/CommonSubexpressionEliminator.cpp [hunk 3]
```diff
@@ -75,13 +75,13 @@ void CommonSubexpressionEliminator::visit(Expression& _e)
 	if (holds_alternative<Identifier>(_e))
 	{
 		Identifier& identifier = std::get<Identifier>(_e);
-		YulString name = identifier.name;
-		if (m_value.count(name))
+		YulString identifier_name = identifier.name;
+		if (m_value.count(identifier_name))
```
Comment (leonardoalt) [inline libyul/optimiser/CommonSubexpressionEliminator.cpp]: Camel style

File: libyul/optimiser/DataFlowAnalyzer.cpp [hunk 4]
```diff
@@ -41,30 +41,30 @@ using namespace yul;
 
 void DataFlowAnalyzer::operator()(ExpressionStatement& _statement)
 {
-	if (auto vars = isSimpleStore(dev::eth::Instruction::SSTORE, _statement))
+	if (auto sstore_vars = isSimpleStore(dev::eth::Instruction::SSTORE, _statement))
```
Comment (leonardoalt) [inline libyul/optimiser/DataFlowAnalyzer.cpp]: Camel

File: libyul/optimiser/Rematerialiser.cpp [hunk 5]
```diff
@@ -71,20 +71,20 @@ void Rematerialiser::visit(Expression& _e)
 	if (holds_alternative<Identifier>(_e))
 	{
 		Identifier& identifier = std::get<Identifier>(_e);
-		YulString name = identifier.name;
-		if (m_value.count(name))
+		YulString identifier_name = identifier.name;
```
Comment (leonardoalt) [inline libyul/optimiser/Rematerialiser.cpp]: Camel

File: libyul/optimiser/VarNameCleaner.cpp [hunk 6]
```diff
@@ -87,9 +87,9 @@ void VarNameCleaner::renameVariables(vector<TypedName>& _variables)
 
 void VarNameCleaner::operator()(Identifier& _identifier)
 {
-	auto name = m_translatedNames.find(_identifier.name);
-	if (name != m_translatedNames.end())
-		_identifier.name = name->second;
+	auto identifier_name = m_translatedNames.find(_identifier.name);
```
Comment (leonardoalt) [inline libyul/optimiser/VarNameCleaner.cpp]: Camel

File: test/libsolidity/ASTJSONTest.cpp [hunk 7]
```diff
@@ -75,7 +75,6 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 	file.open(m_astFilename);
 	if (file)
 	{
```
Comment (leonardoalt) [inline test/libsolidity/ASTJSONTest.cpp]: Brackets can now be removed

File: test/libsolidity/ASTJSONTest.cpp [hunk 8]
```diff
@@ -84,7 +83,6 @@ ASTJSONTest::ASTJSONTest(string const& _filename)
 	file.open(m_legacyAstFilename);
 	if (file)
 	{
```
Comment (leonardoalt) [inline test/libsolidity/ASTJSONTest.cpp]: Here too

File: test/libsolidity/SolidityExpressionCompiler.cpp [hunk 9]
```diff
@@ -131,7 +131,6 @@ bytes compileFirstExpression(
 	for (ASTPointer<ASTNode> const& node: sourceUnit->nodes())
 		if (ContractDefinition* contract = dynamic_cast<ContractDefinition*>(node.get()))
 		{
-			ErrorReporter errorReporter(errors);
```
Comment (leonardoalt) [inline test/libsolidity/SolidityExpressionCompiler.cpp]: Maybe the idea here was to have a fresh one?
Not sure

**Comment by axic** [comment]
> If we really want to, I'd manage to cope, but to me personally that feels much more annoying than helpful... what mistake did that reveal?

@ekpyron I can't fully remember, but it did find this: https://github.com/ethereum/solidity/pull/8013#discussion_r359476670

**Comment by ekpyron** [comment]
> > If we really want to, I'd manage to cope, but to me personally that feels much more annoying than helpful... what mistake did that reveal?
> 
> @ekpyron I can't fully remember, but it did find this: [#8013 (comment)](https://github.com/ethereum/solidity/pull/8013#discussion_r359476670)

That's not *really* an error, though :-)... I mean it's nicer after the change, but it wasn't wrong before...

**Comment by axic** [comment]
This commit 3ecbcbb001  works. I'd prefer merging that and deal with shadowing separately. Any comments?

**Comment by axic** [comment]
@ekpyron btw if shadowing is annoying, don't even dare to look at implicit conversion warnings 😨

**Comment by ekpyron** [comment]
> @ekpyron btw if shadowing is annoying, don't even dare to look at implicit conversion warnings fearful

Haha, yeah, I can imagine, but warnings on narrowing conversions I'd much rather have than on shadowing :-). So I'd go ahead without that commit, but yeah - feel free to vote me down on the shadowing stuff, if you think it's worthwhile :-) - I could live with it.

**Comment by chriseth** [comment]
Needs rebase.

**Comment by chriseth** [comment]
Will compile on my system and rebase.

**Comment by chriseth** [comment]
Decision from call: do not enable shadowing warning.

**Comment by axic** [comment]
@chriseth @ekpyron @leonardoalt rebased and removed shadowing

**Comment by axic** [comment]
The travis failure is the bytecode uploader.

**Review by chriseth — APPROVED** [review]

**Pull request merged**